gobblin-utility/src/main/java/org/apache/gobblin/util/filesystem/PathAlterationObserver.java

```diff
@@ -187,6 +186,14 @@ public void checkAndNotify()
     for (final PathAlterationListener listener : listeners.values()) {
       listener.onStop(this);
     }
+
+    if (this.changeApplied) {
+      for (final PathAlterationListener listener : listeners.values()) {
+        // Fire onCheckDetectedChange to notify when one check contains any number of changes
+        listener.onCheckDetectedChange();
+      }
+    }
+    this.changeApplied = false;
```

Comment:
unless a semantic difference, I suggest to init to false as the first line of the method, rather than the last (which I believe to be leaving it for the next subsequent invocation).